src/Extensions/UsedOnTableExtension.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace SilverStripe\LinkField\Extensions;
+
+use DNADesign\Elemental\Models\BaseElement;
+use SilverStripe\Core\Extension;
+use SilverStripe\LinkField\Models\FileLink;
+use SilverStripe\ORM\DataObject;
+
+class UsedOnTableExtension extends Extension
+{
+    public function updateUsageAncestorDataObjects(array &$ancestorDataObjects, DataObject $dataObject): void
+    {
+        if (!is_a($dataObject, FileLink::class)) {
+            return;
+        }
+        $owner = $dataObject->Owner();
```

Comment:
```suggestion
        $owner = $dataObject->Owner();
        if (!$owner || !$owner->exists()) {
            return;
        }
```
Unless I'm misunderstanding, we probably don't want to add a null or a dummy record into the ancestor array.

Note that while the `exists()` call is currently redundant in this context, it's good practice and protects against possible changes in the future.

Reply:
> Note that while the exists() call is currently redundant

Should just deprecate the method then rather than using it?

Reply:
Updated